macro expansion producing 'defined' has undefined behavior [-Wexpansion-to-defined] with Apple clang 17.0.0

```
./../m4ri/mzd.h:1002:5: warning: macro expansion producing 'defined' has undefined behavior [-Wexpansion-to-defined]
 1002 | #if __M4RI_HAVE_SSE2
      |     ^
./../m4ri/m4ri_config.h:7:28: note: expanded from macro '__M4RI_HAVE_SSE2'
    7 | #define __M4RI_HAVE_SSE2                (defined(__SSE2__) && __SSE2__)
      |                                          ^
```
the latter `#define` should be using if/then/else instead. 

I'll fix it and do a PR

--- a/configure.ac
+++ b/configure.ac
@@ -60,9 +60,9 @@ AS_IF([test "x$enable_sse2" != "xno"], [
 ])
 
 AS_IF([test "x$enable_sse2" = "xno"],
-    [M4RI_HAVE_SSE2=0],
-    [M4RI_HAVE_SSE2="(defined(__SSE2__) && __SSE2__)"])
-AC_SUBST(M4RI_HAVE_SSE2)
+    [M4RI_SSE2_ENABLED=0],
+    [M4RI_SSE2_ENABLED=1])
+AC_SUBST(M4RI_SSE2_ENABLED)
 
 AC_ARG_WITH(papi,
     AS_HELP_STRING([--with-papi@<:@=PATH@:>@], [The PAPI install prefix, if configure can't find it.]),

--- a/m4ri/m4ri_config.h.in
+++ b/m4ri/m4ri_config.h.in
@@ -4,7 +4,11 @@
 // Defines determined during configuration of m4ri.
 #define __M4RI_HAVE_MM_MALLOC		@M4RI_HAVE_MM_MALLOC@
 #define __M4RI_HAVE_POSIX_MEMALIGN	@M4RI_HAVE_POSIX_MEMALIGN@
-#define __M4RI_HAVE_SSE2		@M4RI_HAVE_SSE2@
+#define __M4RI_HAVE_SSE2		0
+#if @M4RI_SSE2_ENABLED@ && defined(__SSE2__) && __SSE2__
+#undef __M4RI_HAVE_SSE2
+#define __M4RI_HAVE_SSE2		1
+#endif
 #define __M4RI_HAVE_OPENMP		@M4RI_HAVE_OPENMP@
 #define __M4RI_CPU_L1_CACHE		@M4RI_CPU_L1_CACHE@
 #define __M4RI_CPU_L2_CACHE		@M4RI_CPU_L2_CACHE@

